src/applications/appeals/10182/utils/replace.js

```diff
@@ -45,3 +47,17 @@ export const replaceSubmittedData = text =>
     (resultingText, transformer) => transformer(resultingText),
     text || '',
   );
+
+/**
+ * Change a date string with no leading zeros (e.g. 2020-1-2) into a date with
+ * leading zeros (e.g. 2020-01-02) as expected in the schema
+ * @param {String} dateString YYYY-M-D or YYYY-MM-DD date string
+ * @returns {String} YYYY-MM-DD date string
+ */
+export const fixDateFormat = (dateString = '') => {
+  if (dateString.length === 10) {
+    return dateString;
+  }
+  const { day, month, year } = parseISODate(dateString);
+  return `${year}-${month.padStart(2, '0')}-${(day || '').padStart(2, '0')}`;
```

Comment:
wondering if the `(day || '')` should instead be `(day || '1')` ?

If I gave it a date without a day right now like 2023-1, I think it would give me 2023-01-00, which wouldn't be a valid value, right?

Not sure it is a valid case that would ever be encountered, but otherwise this all looks good to me.

Reply:
Oh right, good catch!

Reply:
Actually, I'm not going to change this... this function is used by the form `validateDate` function which expects a zero. If I do implement this change, then I should also catch an undefined month (using `(month || '1')` and this causes a value of `2000--` to return `2000-01-01`

Reply:
But it actually returns `2000-00-00` because of the `parseISODate` platform function. I spent _way too much time_ trying to change that function to return an empty string instead of a `0`, but ended up breaking other tests, etc, etc...